src/analyses/base.ml

```diff
@@ -2026,6 +2026,80 @@ struct
         M.warn ~category:(Behavior (Undefined InvalidMemoryDeallocation)) ~tags:[CWE 761] "Free of memory not at start of buffer in function %s for pointer %a" special_fn.vname d_exp ptr
     | _ -> M.warn ~category:MessageCategory.Analyzer "Pointer %a in function %s doesn't evaluate to a valid address." d_exp ptr special_fn.vname
 
+  (* Get the size of the smallest memory that dest points-to *)
+  let get_min_size_of_dest ctx dest =
+    let intdom_of_int x =
+      ID.of_int (Cilfacade.ptrdiff_ikind ()) (Z.of_int x)
+    in
+    let points_to_heap_only =
+      match ctx.ask (Queries.MayPointTo dest) with
+      | a when not (AD.is_top a) ->
+        AD.for_all (function
+            | Addr (v, _) -> ctx.ask (Queries.IsHeapVar v)
+            | _ -> false
+          ) a
+      | _ -> false
+    in
+    if points_to_heap_only then
+      (* Ask for BlobSize from the base address (the second field set to true) in order to avoid BlobSize giving us bot *)
+      ctx.ask (Queries.BlobSize {exp = dest; base_address = true})
```

Comment:
Does this not ignore offsets in `dest`? If I have a pointer to the end of something that is `80` bytes large, I cannot copy `80` bytes into it...

Reply:
True, yes. I'll make sure to set `base_address` to `false`.

Reply:
Fixed in https://github.com/goblint/analyzer/pull/1197/commits/2534945eec78d99658cc2bf5b420af35580b7fe3